src/execution/operator/join/physical_right_delim_join.cpp

```diff
@@ -0,0 +1,141 @@
+	auto &lstate = input.local_state.Cast<RightDelimJoinLocalState>();
+
+	OperatorSinkInput join_sink_input {*join->sink_state, *lstate.join_state, input.interrupt_state};
+	join->Sink(context, chunk, join_sink_input);
+
+	OperatorSinkInput distinct_sink_input {*distinct->sink_state, *lstate.distinct_state, input.interrupt_state};
+	distinct->Sink(context, chunk, distinct_sink_input);
+
+	return SinkResultType::NEED_MORE_INPUT;
+}
+
+SinkCombineResultType PhysicalRightDelimJoin::Combine(ExecutionContext &context,
+                                                      OperatorSinkCombineInput &input) const {
+	auto &lstate = input.local_state.Cast<RightDelimJoinLocalState>();
+
+	OperatorSinkCombineInput join_combine_input {*join->sink_state, *lstate.join_state, input.interrupt_state};
+	join->Combine(context, join_combine_input);
+
+	OperatorSinkCombineInput distinct_combine_input {*distinct->sink_state, *lstate.distinct_state,
+	                                                 input.interrupt_state};
+	distinct->Combine(context, distinct_combine_input);
+
+	return SinkCombineResultType::FINISHED;
+}
+
+SinkFinalizeType PhysicalRightDelimJoin::Finalize(Pipeline &pipeline, Event &event, ClientContext &client,
+                                                  OperatorSinkFinalizeInput &input) const {
+	D_ASSERT(join);
+	D_ASSERT(distinct);
+
+	OperatorSinkFinalizeInput join_finalize_input {*join->sink_state, input.interrupt_state};
+	join->Finalize(pipeline, event, client, join_finalize_input);
+
+	OperatorSinkFinalizeInput distinct_finalize_input {*distinct->sink_state, input.interrupt_state};
+	distinct->Finalize(pipeline, event, client, distinct_finalize_input);
+
+	return SinkFinalizeType::READY;
+}
+
+//===--------------------------------------------------------------------===//
+// Pipeline Construction
+//===--------------------------------------------------------------------===//
+void PhysicalRightDelimJoin::BuildPipelines(Pipeline &current, MetaPipeline &meta_pipeline) {
+	op_state.reset();
+	sink_state.reset();
+
+	auto &child_meta_pipeline = meta_pipeline.CreateChildMetaPipeline(current, *this);
+	child_meta_pipeline.Build(*children[0]);
+
+	D_ASSERT(type == PhysicalOperatorType::RIGHT_DELIM_JOIN);
+	// recurse into the actual join
+	// any pipelines in there depend on the main pipeline
+	// any scan of the duplicate eliminated data on the LHS depends on this pipeline
+	// we add an entry to the mapping of (PhysicalOperator*) -> (Pipeline*)
+	auto &state = meta_pipeline.GetState();
+	for (auto &delim_scan : delim_scans) {
+		state.delim_join_dependencies.insert(
+		    make_pair(delim_scan, reference<Pipeline>(*child_meta_pipeline.GetBasePipeline())));
+	}
+
+	// The rest of this function replicates the PhysicalJoin::BuildJoinPipelines,
+	// without building the RHS because we already build that in the Sink phase of this operator
+	state.AddPipelineOperator(current, *join);
```

Comment:
Can we re-use the code from `PhysicalJoin::BuildJoinPipelines` and skip the build of the RHS with a boolean or so?